CLAUDE.md

```diff
@@ -0,0 +1,149 @@
+# CLAUDE.md - EigenDA
+
+> **Purpose** – This file is the onboarding manual for every AI assistant (Claude, Cursor, GPT, etc.) and every
+> human who edits this repository. It encodes our coding standards, guard rails, and workflow tricks.
+
+---
+
+## 1. Non-negotiable golden rules
+
+| #:  | AI *must NOT* do                                                                                                                                   |
+|-----|----------------------------------------------------------------------------------------------------------------------------------------------------|
+| G-0 | Write changes or use tools when you are not sure about something project-specific, or if you don't have context for a particular feature/decision. |
+| G-1 | Touch test files, specs, or generated files without EXPLICIT permission. **Humans own tests and specs.**                                           |
```

Comment:
what does "touch" mean here? Should we not want the AI to run `make protoc` or `make compile-contracts` to update the generated files?

Reply:
I meant "directly modify". Will fix